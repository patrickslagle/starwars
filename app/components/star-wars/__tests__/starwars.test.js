import React from 'react';
import Adapter from 'enzyme-adapter-react-15';
import { shallow, configure } from 'enzyme';
import App from '../app.jsx';
import FooterNav from '../footer-nav.jsx';

configure({ adapter: new Adapter() });

test('App component renders properly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

test('FooterNav component renders properly', () => {
  const fetchPeopleMock = jest.fn();
  const wrapper = shallow(
    <FooterNav 
      fetchPeople={fetchPeopleMock}
      loadingPeople={false}
      next="https://swapi.co/api/people/?page=3"
      prev="https://swapi.co/api/people/?page=1"
    />,
  );
  expect(wrapper).toMatchSnapshot();
});