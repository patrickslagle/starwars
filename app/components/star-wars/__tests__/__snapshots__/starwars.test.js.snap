// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders properly 1`] = `
<div
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h1>
    Star Wars People
  </h1>
  <SearchPeople
    fetchPeople={[Function]}
  />
  <div
    id="people-container"
  />
  <FooterNav
    fetchPeople={[Function]}
    loadingPeople={true}
    next={null}
    prev={null}
  />
</div>
`;

exports[`FooterNav component renders properly 1`] = `
<footer>
  <button
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Prev
  </button>
  <button
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</footer>
`;
